cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/Hive12ExploreService.java

```diff
@@ -129,4 +135,29 @@ protected OperationHandle doExecute(SessionHandle sessionHandle, String statemen
     throws HiveSQLException, ExploreException {
     return getCliService().executeStatementAsync(sessionHandle, statement, ImmutableMap.<String, String>of());
   }
+
+  @Override
+  protected SessionHandle doOpenHiveSession(Map<String, String> sessionConf) throws HiveSQLException {
+    SessionHandle sessionHandle = super.doOpenHiveSession(sessionConf);
+
+    // HDP Hive 12 only sets SessionState when an operation is executed.
+    // The SessionState is not set during an openSession call.
+    // Hence execute a dummy statement during openSession, so that SessionState gets set.
+    OperationHandle opHandle = null;
+    try {
+      opHandle = getCliService().executeStatement(sessionHandle, "set cdap_tmp_var=1",
+                                                  Collections.<String, String>emptyMap());
+    } catch (Throwable t) {
+      LOG.error("Error setting temporary variable to set session state", t);
```

Comment:
is it ok to ignore the exception? Do we presume that it even if there is an exception, the ``SessionHandle`` is set?